### PR TITLE
[CLOUD-1352] Adding doc for Launchpad's Azure access token passthrough mode and for using a client cert

### PR DESCRIPTION
**Pull request by paulplace**
I added two new pages for configuring Launchpad/Stardog to work with Azure, one for configuring access token passthrough mode (as implemented in CLOUD-1278) and one for configuring Launchpad to present a client certificate as its credential when authenticating itself to Azure (as implemented in CLOUD-1332).

I also removed references to the `PASSWORD_AUTH_ENABLED` config variable, per CLOUD-1328.

### CODE REVIEW

File: azure/access-token-passthrough-mode.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+- Docker Compose installed
+- Two registered applications with the [Microsoft Identity Platform](https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app#register-an-application), one for Launchad and one for the Stardog server. More details are provided below.
+- A Stardog server running locally on port `5820` and configured to accept access tokens from the registered Stardog Azure AD application. See [Stardog Server Requirements](#stardog-server-requirements) for additional info.
+
+   > **Note:**
+   > If you have a Stardog server running elsewhere (locally or not), this is fine, just modify the `STARDOG_INTERNAL_ENDPOINT` and `STARDOG_EXTERNAL_ENDPOINT` in the [`.env`](.env) file as needed.
+
+### How to Register the Launchpad Application
+
+Here are steps to create and register Launchpad as a Microsoft Application. Note that these are almost identical to the steps in the [basic mode example](./README.md#how-to-register-the-application), except that the `Directory.Read.All` scope is not required for the Launchpad registered application when using access token passthrough mode.
```
Comment (PaulJackson123) [inline azure/access-token-passthrough-mode.md:49]: Should you indicate here that you'll be registering Launchpad as a web application, in the same way that you indicated that Stardog would be registered as a Mobile and desktop application?

**Review by PaulJackson123 — CHANGES_REQUESTED** [review]
Very nice. Will be helpful to me the next time I need to set this up.

I made a couple comments around how the access key isn't sent from Stardog to the data source directly, rather it's exchanged through the on-behalf-of flow.

It took me a bit to understand what was meant by "passthrough". In this context it means the access token that is obtains by launchpad from Azure AD is used to authenticate with Stardog. I was confused because we also use that term for VG passthrough. Not necessarily requesting a change.

File: README.md [hunk 2]
```diff
@@ -139,32 +139,36 @@ To secure communications between Stardog and Launchpad, see [SSL Cert Verificati
...
+| **Environment Variable**               | **Default**          | **Required**                              | **Description**                                                                                                                                                                                                                                                                                                                                                     |
+| -------------------------------------- | -------------------- | ----------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
```
[the remark this answers is not in the document]
Comment (paulplace) [reply README.md:143]: I don't _think_ so. I played with the formatting a bit, but I couldn't really get the markdown table formatter to do things differently.

File: azure/README.md [hunk 3]
```diff
@@ -34,9 +38,9 @@ Diagram demonstrating the flow described above:
 sequenceDiagram
   Launchpad->>Azure AD: Successful user authentication
   Azure AD->>Launchpad: Azure AD JWT returned
-  Note over Azure AD,Launchpad: Launchpad saves profile information <br> contained in Azure AD JWT in a session and discards it.
+  Note over Azure AD,Launchpad: Launchpad saves profile information <br> from Azure's JWT and discards it.
   Launchpad->>Stardog: Stardog API requests with Launchpad JWT
-  Note over Launchpad,Stardog: Launchpad generates its JWTs Stardog server is configured to accept using information contained in the session.
+  Note over Launchpad,Stardog: Launchpad generates a JWT that the Stardog server is <br> configured to accept, using information from Azure's JWT.
```
[the remark this answers is not in the document]
Comment (paulplace) [reply azure/README.md:43]: I will re-word.

File: azure/access-token-passthrough-mode.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
+# Azure AD Example (Access Token Passthrough mode)
+
+This example builds on the [basic mode example](./README.md), adding support for access token passthrough mode, wherein Launchpad requests an identity token from Azure AD for the purpose of authenticating the user, but it also requests an access token that it passes through to the Stardog server. The Stardog server can then use the credentials of the logged-in user when connecting to a data source for the purpose of querying a virtual graph. In this example, the data source is hosted in Databricks.
```
[the remark this answers is not in the document]
Comment (paulplace) [reply azure/access-token-passthrough-mode.md:3]: Thank you for the clarifying verbiage. I will use it.

File: azure/access-token-passthrough-mode.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+The Pass-Through Authentication mode for Stardog is documented [here](https://docs.stardog.com/virtual-graphs/data-sources/passthrough-authentication). Note that this feature was added in the [9.1.0](https://docs.stardog.com/release-notes/stardog-platform#910-release-2023-07-06) release of the Stardog Platform.
+
+## A Little More Detail
+
+In [basic mode](./README.md), Launchpad uses Azure AD as an OIDC identity provider, requesting an identity token for an authenticated user. Launchpad receives a JSON Web Token (JWT) from Azure AD, extracts some information from the token, and then constructs a new JWT that is used to interact with Stardog.
+
+In **access token passthrough mode**, Launchpad requests both an identity token and an access token from Azure AD for an authenticated user. Launchpad uses the identity token to verify that the user has properly authenticated with Azure AD, and then the identity token is discarded. The access token is used as-is in all interactions with the Stardog server (both from Launchpad and from the Stardog Applications). This access token passthrough mode allows Stardog to act on behalf of the Azure AD user when connecting to a data source.
```
[the remark this answers is not in the document]
Comment (paulplace) [reply azure/access-token-passthrough-mode.md:11]: I added your more detailed explanation above, so I will leave this as-is.

**Pull request merged**